app/src/main/java/com/epfl/drawyourpath/path/RunPopupAdapter.kt

```diff
@@ -0,0 +1,41 @@
+package com.epfl.drawyourpath.mainpage.fragments.helperClasses
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.epfl.drawyourpath.R
+import com.epfl.drawyourpath.path.Run
+
+class RunPopupAdapter(private var runs: List<Run>, private val runClickListener: (Run) -> Unit) :
+    RecyclerView.Adapter<RunPopupAdapter.RunPopupViewHolder>() {
+
+    // ViewHolder class for displaying each run
+    class RunPopupViewHolder(itemView: View, val runClickListener: (Run) -> Unit) : RecyclerView.ViewHolder(itemView) {
+        val runTitle: TextView = itemView.findViewById(R.id.runTitle)
+
+        fun bind(run: Run) {
+            runTitle.text = run.getDate() + " - " + run.getDistance() + " km"

```

Comment:
A bit cleaner to use Kotlin's string escaping.

```suggestion
            runTitle.text = "${run.getDate()} - ${run.getDistance()}km"

```